### ISSUE TITLE
select_related doesn't work...

### ISSUE THREAD

**Issue by ghost** [hmarr/mongoengine#450]
I'm a newbie for django and mongoengine..
so I'm not sure my code is correct on this issue..

```
class Example(Document):
    member = ReferenceField(Members)

query = Example.objects().select_related(max_depth=0)
```

should not query variable have member field? so that when I code

```
query.member.name ---> None
```

but the result would be the member's name like Simson..

In my real application, I don't want to use dereference.. even though the class(Document) has ReferenceField 
nor I want to use "only()", "exclude" methods..

How can I make mongoengine not use dereference even though the class(Document) has its own ReferenceField?

(I just implemented ReferenceField for the future so as to use it one day.. but not now)

**Comment by rozza** [comment]
To do this just provide a string for the object id and don't use ReferenceField - at the top level they will always dereference.

**Comment by rozza** [comment]
Perhaps a lazy flag for the future (default to true).

**Comment by ghost** [comment]
oh~~ "at the top level they will always dereference" !!!
oh~~~~

**Comment by rozza** [comment]
Yeah, the aim was to stop all individual ReferenceFields racking up queries...  But you removing all control over that is also bad.
